tornado-drivers/spirv/src/main/java/uk/ac/manchester/tornado/drivers/spirv/SPIRVDeviceContext.java

```diff
@@ -149,6 +162,14 @@ public int getDriverIndex() {
         return TornadoRuntimeProvider.getTornadoRuntime().getBackendIndex(SPIRVBackendImpl.class);
     }
 
+    private boolean isDeviceContextLevelZero() {
```

Comment:
What if this is done in the concrete instance:

a method within the `SPIRVLevelZeroPowerMetric` and `SPIRVOCLPowerMetric` classes that return true or false accordingly.

Reply:
As discussed internally, we will pause this change and add it as part of a refactoring process that will take place later.